08-transport.Rmd

```diff
@@ -300,26 +322,34 @@ Desire lines contain only two vertices (their beginning and end points) but rout
 There are many advantages of using remote routing services, including the fact that they can be up-to-date, have global coverage, and run on a set-up that was designed for the job, explaining this section's focus on online routing services.
-Before proceeding, however, you should be aware of the disadvantages of online routing services: they can be slow due to their reliance on internet connections and expensive.
+Before proceeding, however, you should be aware of the disadvantages of online routing services: they are slow or even inaccessible when accessed with a bad internet connection and can be expensive, i.e., not all services are free of charge.
+For instance, the Google routing API sets the limit to 2500 free queries per day ^[`ggmap::route()` lets you use Google's routing API through R.].
 <!-- Todo: add link to Mark's presentation on dodgr vs Google routing costs (RL) -->
 
-A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).
-Instead of routing *all* desire lines generated in the previous section, which would be time and memory-consuming, we will focus attention on only the desire lines of policy interest.
+A popular and free (for limited uses) routing service is the Open Source Routing Machine ([OSRM](http://project-osrm.org/)).^[Another popular OSM routing API is https://openrouteservice.org/.]
+Instead of routing *all* desire lines generated in the previous section, which would be time and memory-consuming, we will focus on the desire lines of policy interest.
 The benefits of cycling trips are greatest when they replace car trips.
 Clearly not all car trips can realistically be replaced by cycling, but 5 km Euclidean distance (or around 6-8 km of route distance) is easily accessible within 30 minutes for most people.
 Based on this reasoning
 <!-- Todo: add references supporting this (RL) -->
-we will only route desire lines along which a high (300+) number of car trips take place that are up to 5 km in distance:
+we will only route desire lines along which a high (300+) number of car trips take place that are up to 5 km in distance.
 
+<!-- st_length works with sf 0.6.0 but not with 0.6.1 -->
 ```{r, message=FALSE}
 desire_lines$distance = as.numeric(st_length(desire_lines))
-desire_carshort = filter(desire_lines, car_driver > 300, distance < 5000)
+desire_carshort = dplyr::filter(desire_lines, car_driver > 300 & distance < 5000)
 route_carshort = line2route(desire_carshort, route_fun = route_osrm)
 ```
 
+`st_length()` determines the length of a linestring, and falls into the distance relations category (see also section \@ref(distance-relations)).
+<!-- could you pls explain a bit more in detail what kind of input the line2route function expects. In the help it says it expects from and to coordinates. But here you use a simple feature. Since this is a line I am not sure what coordinates the function uses. -->
```

Comment:
Could you pls explain a bit more in detail what kind of input the line2route function expects. In the function help it says it expects from and to coordinates. But here you use a simple feature. Since this is a line I am not sure what coordinates the function uses.

Reply:
OK yes the documentation of `stplanr::line2route()` needs updating. I have been planning a generic `route()` function that takes a wide variety of inputs (coordinates, lines, text strings of addresses) and creates routes accordingly. Seem like a good idea to do that and include that here?

Here's the beginning of it btw: https://github.com/ropensci/stplanr/blob/master/R/route.R